pkg/sql/parsers/tree/token_restore.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2023 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tree
+
+import (
+	"fmt"
+	"github.com/matrixorigin/matrixone/pkg/sql/parsers/dialect"
+	"strings"
+)
+
+func NewFmtCtxWithFlag(dialectType dialect.DialectType, flags RestoreFlags) *FmtCtx {
```

Comment:
Unnecessary to add new func, add new opt in NewFmtCtx(dialectType dialect.DialectType, opts ...FmtCtxOption)

Reply:
I have discussed the issue of this variable parameter with Xie Zexiong and tried to modify it, but it is currently not compatible, @iamlinjunhong